test/e2e/retina_perf_test.go

```diff
@@ -0,0 +1,66 @@
+package retina
+
+import (
+	"crypto/rand"
+	"math/big"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/microsoft/retina/test/e2e/common"
+	"github.com/microsoft/retina/test/e2e/framework/types"
+	jobs "github.com/microsoft/retina/test/e2e/jobs"
+	"github.com/stretchr/testify/require"
+)
+
+// TestE2ERetina tests all e2e scenarios for retina
+func TestPerfRetina(t *testing.T) {
+	curuser, err := user.Current()
+	require.NoError(t, err)
+
+	clusterName := curuser.Username + common.NetObsRGtag + strconv.FormatInt(time.Now().Unix(), 10)
+
+	subID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	require.NotEmpty(t, subID)
+
+	location := os.Getenv("AZURE_LOCATION")
+	if location == "" {
+		var nBig *big.Int
+		nBig, err = rand.Int(rand.Reader, big.NewInt(int64(len(common.AzureLocations))))
+		if err != nil {
+			t.Fatalf("Failed to generate a secure random index: %v", err)
+		}
+		location = common.AzureLocations[nBig.Int64()]
+	}
+
+	cwd, err := os.Getwd()
+	require.NoError(t, err)
+
+	// Get to root of the repo by going up two directories
+	rootDir := filepath.Dir(filepath.Dir(cwd))
+
+	chartPath := filepath.Join(rootDir, "deploy", "legacy", "manifests", "controller", "helm", "retina")
+	kubeConfigFilePath := filepath.Join(rootDir, "test", "e2e", "test.pem")
+
+	// CreateTestInfra
+	createTestInfra := types.NewRunner(t, jobs.CreateTestInfra(subID, clusterName, location, kubeConfigFilePath))
+	createTestInfra.Run()
+
+	// Hacky way to ensure that the test infra is deleted even if the test panics
+	defer func() {
+		if r := recover(); r != nil {
+			t.Logf("Recovered in TestE2ERetina, %v", r)
+		}
+		err := jobs.DeleteTestInfra(subID, clusterName, location).Run()
+		if err != nil {
+			t.Logf("Failed to delete test infrastructure: %v", err)
+		}
+	}()
+
+	// Gather benchmark results then install retina and run the performance tests
+	runner := types.NewRunner(t, jobs.RunPerfTest(kubeConfigFilePath, chartPath))
+	runner.Run()
```

Comment:
What are we actually testing here? What assertions of behavior are being made?

Reply:
This test basically generating regression data and not asserting anything.

Reply:
If that be the case, we should just create another binary artifact that runs these perf tests. The testing framework is just getting in the way since it wasn't designed for this purpose.

Reply:
It makes sense to have it as a standalone binary, I would do that refactoring in a follow up in order for this to generate some data for us to analyze and see if we want some other changes.